indigo_api/static/xsl/fo/_hier.xsl

```diff
@@ -0,0 +1,239 @@
+                    <xsl:with-param name="num" select="akn:num"/>
+                  </xsl:call-template>
+                </xsl:when>
+                <xsl:otherwise>
+                  <xsl:value-of select="akn:num"/>
+                </xsl:otherwise>
+              </xsl:choose>
+            </fo:block>
+          </fo:list-item-label>
+          <fo:list-item-body start-indent="{$indent}">
+            <!-- basic units always get a heading; use a non-breaking space if it's missing for alignment -->
+            <fo:block font-weight="bold" font-size="{$fontsize-h3}" keep-with-next="always">
+              <!-- optionally include startQuote character with heading -->
+              <xsl:if test="parent::akn:embeddedStructure and not(preceding-sibling::*) and not(akn:num)">
+                <xsl:call-template name="start-quote">
+                  <xsl:with-param name="quote-char" select="parent::akn:embeddedStructure/@startQuote"/>
+                  <xsl:with-param name="num" select="akn:num"/>
+                </xsl:call-template>
+              </xsl:if>
+              <xsl:apply-templates select="akn:heading"/>
+              <xsl:if test="not(akn:heading)">
+                <xsl:text>&#160;</xsl:text>
+              </xsl:if>
+            </fo:block>
+            <xsl:if test="akn:subheading">
+              <fo:block font-weight="bold" font-size="{$fontsize-h4}" keep-with-next="always">
+                <xsl:apply-templates select="akn:subheading"/>
+              </fo:block>
+            </xsl:if>
+            <!-- basic unit content always goes below the num -->
+            <xsl:apply-templates select="./*[not(self::akn:num|self::akn:heading|self::akn:subheading)]"/>
+            <!-- if this element doesn't have content, force an empty block for the heading to keep with next -->
+            <xsl:if test="not(./*[not(self::akn:num|self::akn:heading)]//text())">
+              <fo:block/>
+            </xsl:if>
+          </fo:list-item-body>
+        </fo:list-item>
+      </fo:list-block>
+    </fo:block-container>
+  </xsl:template>
+
+  <!-- all other hierarchical and numbered elements
+   - startQuote if it's the first element in the quote
+   - number to the side (if present)
+   - heading in bold (if present)
+   - subheading in bold (if present)
+   - content: either lined up with num or below heading/subheading
+
+   note: using list items means that when a parent element doesn't have content, the nums line up
+   e.g. (1) (a) (i) content -->
+  <xsl:template match="akn:alinea|akn:indent|akn:blockList/akn:item|akn:level|akn:list|akn:paragraph|akn:point|
+                       akn:proviso|akn:sublist|akn:subparagraph|akn:subrule|akn:subsection|akn:transitional|
+                       akn:li[not(parent::akn:ul[@class='notice-list'] or parent::akn:ol[@class='amendment-list'])]">
+    <!-- use a block container to retain relative indentation (nesting depth) -->
+    <fo:block-container>
```

Comment:
Similar to above, maybe have a generic `hier-element` template which this template defers to.